### PR TITLE
Make gcc cleanup helper calls public in most of our sd-xyz APIs

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/systemd/_sd-common.h [hunk 1]
```diff
@@ -75,4 +75,11 @@
 #  endif
 #endif
 
+#define _SD_DEFINE_POINTER_CLEANUP_FUNC(type, func)             \
+        static inline void func##p(type **p) {                  \
+                if (*p)                                         \
+                        func(*p);                               \
+        }                                                       \
+        struct __useless_struct_to_allow_trailing_semicolon__
```
Comment (dvdhrm) [inline src/systemd/_sd-common.h:83]: Rather `struct _sd_useless_struct_to_allow_trailing_semicolon`?

But.. I probably cannot give a damn ;)

**Comment by dvdhrm** [comment]
Weird mix of unrelated commits, but they look all fine. I didn't check all the replacements, but if semaphore is happy, I'm happy.

**Comment by dvdhrm** [comment]
Wait, why did you rename `sd_lldp_free` to `sd_lldp_unref`? That doesn't make any sense. If no ref-counting is used, it should remain `free`.

**Comment by poettering** [comment]
@dvdhrm well, it's a question of future-proofing things. Quite a few APIs (at least initially) we had, only started with an _unref() call, that just freed things, without any matching _ref() call, simply to keep the option open to add proper refcounting later on. In this case, if lldp ever becomes a public API I am pretty sure we want to add proper refcounting to it too, to match the other APIs such as dhcp, dhcp6, ipv4ll, ipv4acd, ndisc and so on, which all use refcounting.

File: src/systemd/_sd-common.h (hunk 1 above)
Comment (poettering) [inline src/systemd/_sd-common.h:83]: Will queue a fix for that in a later PR.

Comment (poettering) [inline src/systemd/_sd-common.h:83]: (in particular as this is not the only place we use such a struct in public APIs)

**Comment by dvdhrm** [comment]
But `sd_event_source` shows that adding ref-counts often just confuses people. In a lot of cases, ref-counting just doesn't make sense, especially if the object requires a unique owner.

Anyway, wrong place to discuss this. I'll merge it as is, now.

**Pull request merged**